markdown/docs/addon-author-guide.md

```diff
@@ -0,0 +1,321 @@
+# Addon Author Guide
+
+If you're the author of an Ember addon, you might need to make some
+changes to ensure that your addon runs correctly in the FastBoot
+environment.
+
+Remember that when updating your addons for FastBoot compatibility,
+your code now must run in both a browser *and* Node.js. Most
+importantly, that means that you cannot just assume that there will
+always be DOM APIs available.
+
+## Don't Break the Boot
+
+The first step towards FastBoot support is ensuring that your addon
+doesn't cause the app to crash when run in Node. You can verify this by
+adding your addon to a new Ember app, adding the FastBoot addon, and
+running the FastBoot server with `ember fastboot`.
+
+If that process results in an error and the server crashing, you know
+that your addon is trying to do something that may work in the browser
+but not in Node.js.
+
+### Common Causes
+
+The most common cause of failure on boot is initialization code that
+relies on the following global objects that aren't available in Node.js:
+
+* `document`
+* `navigator`
+* `window`
+
+For example, many addons may detect features by creating an element via
+`document.createElement()` and checking whether certain properties on it
+exist. Or an addon may try to detect the height of the viewport via
+`window.innerHeight`, but because there is no viewport conceptually in
+Node, this value is not set.
+
+### Tracking It Down
+
+FastBoot works by turning an Ember application and all of its
+dependencies (including your addon) into a single bundle of JavaScript.
```

Comment:
For me, em dashes denote an abrupt change of thought, or an interjection. Parentheses denote an explanatory or clarifying remark.